props/prop_base.h

```diff
@@ -958,6 +958,25 @@ class PropBase : CommandParser, Looper, protected SaberBase {
         }
         break;
       }
+      case SaberBase::COLOR_CHANGE_MODE_ZOOMED: {
+#define ZOOM_ANGLE (M_PI * 2 / 2000)
```

Comment:
For 2000 steps per revolution, we need code that can calculate how many steps to go instead of just going one step at a time.

Reply:
COLOR_CHANGE_ZOOMED is used while the PWR button is held after using COLOR_CHANGE_SMOOTH to allow user to dial in color better, it's not a standalone mode.  When PWR is released we ToggleColorChangeMode() to then SaveVariation.  It's intended to go from SMOOTH to STEPPED once the user has nearly the right color to allow for fine-tuning before saving.  I did a lot of testing to get the 2000 value so that it both gave you the ability to still fine-tune but also made it easy to stay in range of selected color with button depressed.

Reply:
I have no problem with the 2000 value, but I don't think the ticked code is very suitable.
Instead I think we should do something like:

float a = fusor.angle2() - current_tick_angle_; 
if (a > M_PI) a-=M_PI*2;
if (a < -M_PI) a+=M_PI*2;
int steps = (int)floor(fabs(a) / ZOOM_ANGLE - 0.3);
if (steps < 0) steps = 0;
if (a < 0) steps = -steps;
current_tick_angle_ += ZOOM_ANGLE * steps;
SaberBase::SetVariation(0x7fff & (SaberBase::GetVariation() + steps));

It should be pretty similar, but be more consistent if you turn back and forth I think.

Reply:
ok, I will test this and make changes, thanks.

Reply:
SaberBase::SetVariation(0x7fff & (SaberBase::GetCurrentVariation() + steps));

this is not working as intended but unable to get it work properly.  With ZOOMED you are in COLOR_CHANGE_SMOOTH to rotate through all colors, then you press the Pwr button and it switches to ZOOMED as long as Pwr is pressed letting you fine tune, when you release that color is then saved.  The above is resetting the color when Pwr is pressed back to original and then adjusting instead of allowing adjustment in the color you had rotated to before pressing the button.  I tried switching to UpdateVariation but it is not working so not sure how to modify this to work as original code did.

Reply:
fixed, should be good now.